### PR TITLE
enums_types / it-academy.by / JC1-117-22 / Aliaksei Kharko / Done tasks 01-07

### PR DESCRIPTION
**Pull request by KharkoAO**
## Pull request state

Actual status for PR:
- [x] implemented
- [ ] reviewed
- [ ] fixed
- [ ] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/enum_types/SeasonsEnum.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    SPRING(7),
+    SUMMER(18) {
+        @Override
+        public String getDescription() {
+            return "Warm season.";
+        }
+    },
+    AUTUMN(7);
+
+    int averageTemperature;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/enum_types/SeasonsEnum.java:14]:
```suggestion
    private final int averageTemperature;
```

File: src/main/java/com/rakovets/course/java/core/practice/enum_types/SeasonsEnum.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                break;
+            case AUTUMN:
+                infoSeason = "I love autumn!";
+                break;
+            default:
+                break;
+        }
+
+        return infoSeason;
+    }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/enum_types/SeasonsEnum.java:45]: move to another class

File: src/main/java/com/rakovets/course/java/core/practice/enum_types/SeasonsEnum.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+        System.out.println(favoriteSeason + " " + favoriteSeason.averageTemperature + " "
+                + favoriteSeason.getDescription());
+
+        System.out.println(favoriteSeason.favoriteSeason(favoriteSeason));
+
+        for (SeasonsEnum season : SeasonsEnum.values()) {
+            System.out.println(season + " " + season.averageTemperature + " " + season.getDescription());
+        }
+    }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/enum_types/SeasonsEnum.java:58]: move to Demo class

**Review by rakovets — CHANGES_REQUESTED** [review]
need to fix

File: src/main/java/com/rakovets/course/java/core/practice/enum_types/SeasonsEnum.java (hunk 1 above)
Comment (KharkoAO) [reply src/main/java/com/rakovets/course/java/core/practice/enum_types/SeasonsEnum.java]: this fixed

File: src/main/java/com/rakovets/course/java/core/practice/enum_types/SeasonsEnum.java (hunk 2 above)
Comment (KharkoAO) [reply src/main/java/com/rakovets/course/java/core/practice/enum_types/SeasonsEnum.java]: this fixed

File: src/main/java/com/rakovets/course/java/core/practice/enum_types/SeasonsEnum.java (hunk 3 above)
Comment (KharkoAO) [reply src/main/java/com/rakovets/course/java/core/practice/enum_types/SeasonsEnum.java]: this fixed

**Pull request closed without merging**